### ISSUE TITLE
birsync can lock forever

### ISSUE THREAD

**Issue by msh100** [JayGoldberg/birsync#1]
After a quick read through the bash script, I suspect that https://github.com/JayGoldberg/birsync/blob/master/birsync.bash#L60 if triggered can cause the lock to stay in place despite rsync exiting (regardless of the exit code).

Suggest this is removed regardless?

**Comment by JayGoldberg** [comment]
Yes, this is intentional. Perhaps I'm not seeing the bug where it will not remove the lock regardless of exit code? If `exit == 0` then we remove the lock file. Otherwise there was an unexpected error. Perhaps you had the wrong source/dest, or the disk is full, etc.

Lockfile is also removed if you exit the process (Ctrl+c),  `cleaup()` is called and removes the lockfile as expected.

If we remove the lockfile in a failed `rsync` run it's conceivable that `rsync` would run in a nearly infinite loop.

Perhaps I can maintain an error count variable where we'll tolerate 3 consecutive failures before leaving the lockfile in place? Or some retry logic with a backoff.

**Comment by msh100** [comment]
I appreciate the lack of wanting to infinitely retry but at the moment it can't retry... at all. If rsync is interrupted for whatever reason nothing will ever progress again.

Simplest solution could just be to add a sleep before removing the lockfile if there's a non-zero exit code from rsync

**Comment by JayGoldberg** [comment]
Correct, it can't retry at all currently. My thinking is that things could go really bad, especially if you're on an embedded system where the farther you fall behind, the more compute that rsync is using to reconcile two dirs, only to hit a condition where the sync can't occur anyway (disk full, etc). I agree that the silent failure is not good (we should email the user or call an HTTP hook), but at least it forces the user to find the root cause of the problem.

**Comment by JayGoldberg** [comment]
I'll open an issue to add retry logic, and also to call an HTTP endpoint if there is an issue.
